ui/src/app/job-details/tabs/timing-diagram/timing-diagram.component.ts

```diff
@@ -20,23 +21,42 @@ export class JobTimingDiagramComponent implements OnInit {
       chartType: 'Timeline',
       dataTable: []
     };
-    this.timelineChart.dataTable.push(['Num', 'Name', 'Start', 'End']);
+    this.timelineChart.dataTable.push([
+      {'type': 'string', 'id': 'Task'},
+      {'type': 'string', 'id': 'Section'},
+      {'type': 'date', 'purpose': 'Start', 'id': 'Start'},
+      {'type': 'date', 'purpose': 'End', 'id': 'End'}
+    ]);
     let counter = 1;
 
     metadata.forEach((task) => {
       if (task.name && task.start) {
-        if (task.end) {
-          this.timelineChart.dataTable.push([counter.toString(), task.name, task.start, task.end]);
+        if (task.executionEvents) {
+          task.executionEvents.forEach((event) => {
+            this.timelineChart.dataTable.push(this.formatRow(task.name, event.name, event));
+          });
         } else {
-          this.timelineChart.dataTable.push([counter.toString(), task.name, task.start, new Date()]);
+          this.timelineChart.dataTable.push(this.formatRow(task.name, task.name, task));
         }
         counter++;
       }
-    })
+    });
+
     this.timelineChart.options = {
-      title: 'Tasks',
-      width: 800,
-      height: counter * 50
+      timeline: {
+        showRowLabels: false,
+        showBarLabels: false
+      },
+      width: this.tabWidth - 65,
```

Comment:
Looks like this only runs on `ngOnInit` so I'm not sure I'd expect this to resize itself properly. Perhaps a percentage based width would work better (if supported?).

Anyways - I can't really test this out, so I'll leave it to you or another reviewer to ensure this is functional

Reply:
It isn't set to re-size, just to figure out how wide to make it when the component is first rendered.